VideoLocker/src/main/java/org/edx/mobile/view/RegisterActivity.java

```diff
@@ -386,4 +459,196 @@ public boolean onCreateOptionsMenu(Menu menu) {
         // Launch screen doesn't have any menu
         return true;
     }
+
+    /**
+     * we can create enum for strong type, but lose the extensibility.
+     * @param socialType
+     */
+    private void showRegularMessage(int socialType){
+        LinearLayout messageLayout = (LinearLayout) findViewById(R.id.message_layout);
+        ETextView messageView = (ETextView) findViewById(R.id.message_body);
+        //we replace facebook and google programmatically here
+        //in order to make localization work
+        String socialTypeString = "";
+        String signUpSuccessString = "";
+        if ( socialType == SocialFactory.TYPE_FACEBOOK ){
+            socialTypeString = getString(R.string.facebook_text);
+            signUpSuccessString = getString(R.string.sign_up_with_facebook_ok);
+        } else {  //google
+            socialTypeString = getString(R.string.google_text);
+            signUpSuccessString = getString(R.string.sign_up_with_google_ok);
+        }
+        StringBuilder sb = new StringBuilder();
+        sb.append( signUpSuccessString.replace(socialTypeString, "<b><strong>" + socialTypeString + "</strong></b>") )
+            .append("<br>").append( getString(R.string.sign_up_with_social_ok) );
+
+        Spanned result = Html.fromHtml(sb.toString());
+        messageView.setText(result);
+        messageLayout.setVisibility(View.VISIBLE);
+       // UiUtil.animateLayouts(messageLayout);
+    }
+
+    private void updateUIOnSocialLoginToEdxFailure(int socialType){
+        //change UI.
+        View signupWith = findViewById(R.id.signupWith);
+        signupWith.setVisibility(View.GONE);
+        View socialPanel = findViewById(R.id.panel_social_layout);
+        socialPanel.setVisibility(View.GONE);
+        ETitleRowView signupWithEmailTitle = (ETitleRowView)findViewById(R.id.signupWithEmailTitle);
+        signupWithEmailTitle.setTitle( getString(R.string.complete_registration) );
+        //help method
+        showRegularMessage(socialType);
+        //populate the field with value from social site
+        populateEmailFromSocialSite(socialType);
+        //hide email and password field
+        for (IRegistrationFieldView field : this.mFieldViews ){
+            String fieldname = field.getField().getName();
+            if ( "password".equalsIgnoreCase(fieldname) ) {
```

Comment:
Since we read the form out of a file, hardcoding "password" here is kind of unfortunate. A third party may use a different form which has a different field name for password. Also, eventually we want to download the form from the internet which means we have even less control. For now it's probably okay, since we're not downloading the file, but we should keep this in mind.